django/utils/numberformat.py

```diff
@@ -18,46 +17,93 @@ def format(number, decimal_sep, decimal_pos=None, grouping=0, thousand_sep='',
         module in locale.localeconv() LC_NUMERIC grouping (e.g. (3, 2, 0)).
     * thousand_sep: Thousand separator symbol (for example ",")
     """
-    use_grouping = (use_l10n or (use_l10n is None and settings.USE_L10N)) and settings.USE_THOUSAND_SEPARATOR
-    use_grouping = use_grouping or force_grouping
-    use_grouping = use_grouping and grouping != 0
+    if force_grouping:
+        pass
+    elif not settings.USE_THOUSAND_SEPARATOR:
+        grouping = 0
+    elif use_l10n is False:
+        grouping = 0
+    elif use_l10n is None and not settings.USE_L10N:
+        grouping = 0
```

Comment:
The main speed-gain I was trying to achieve here was to avoid unnecessary checks (e.g. if `force_grouping` is true, or `USE_THOUSAND_SEPARATOR` is false, there is no point completing further checks). 

I agree that `use_grouping` is more explicit, and clearer. `grouping` would still only be a tuple or an integer. Since the default value of `grouping` is 0, I wasn't really thinking of it as a boolean (although I admit, we do treat it as a boolean when we say `if grouping:`). I removed `use_grouping` just because it was an extra variable to pass around. I tried adding use_grouping back in, and it didn't make much difference to speed (maybe 2% more time, but nothing worth fussing about). I could go either way on this.

Reply:
Perhaps you could use `grouping = None` instead of `grouping = 0`?

While not a more descriptive name, it is perhaps more semantically equivalent to grouping being disabled. Would that satisfy you, @codingjoe?

Reply:
I certainly wouldn't have any objection to using `grouping=None`.